kotest-framework/kotest-framework-concurrency/src/jvmTest/kotlin/io/kotest/framework/concurrency/EventuallySpec.kt

```diff
@@ -15,22 +15,23 @@ import io.kotest.matchers.longs.shouldBeGreaterThanOrEqual
 import io.kotest.matchers.nulls.shouldNotBeNull
 import io.kotest.matchers.shouldBe
 import io.kotest.matchers.string.shouldContain
-import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.DelicateCoroutinesApi
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.newSingleThreadContext
 import kotlinx.coroutines.withContext
 import java.io.FileNotFoundException
 import java.io.IOException
 import java.time.Duration
 import java.util.concurrent.CountDownLatch
-import java.util.concurrent.Executors
 import java.util.concurrent.TimeUnit
 import java.util.concurrent.atomic.AtomicInteger
 
 private fun Int.seconds(): Long = Duration.ofSeconds(this.toLong()).toMillis()
 private fun Int.milliseconds(): Long = this.toLong()
 
 @ExperimentalKotest
+@OptIn(DelicateCoroutinesApi::class)
```

Comment:
Are these methods likely to change in future coroutine updates?
We've definitely been burnt many times by the kotlin sdk changing things on us.

Reply:
In 2018, these were marked `@ObsoleteCoroutinesApi`, thought to be replaced by dispatcher views (https://github.com/Kotlin/kotlinx.coroutines/issues/2919, https://github.com/Kotlin/kotlinx.coroutines/issues/2943), but with the latter becoming available in 2021, `newSingleThreadContext` and `newFixedThreadPoolContext` were "promoted" to `@DelicateCoroutinesApi` (though still "experimental"). Given that history, and the fact that they are used in current documentation, I'd say they'll be here to stay.

(They are marked as delicate because these dispatchers need to be closed, which is now consistently done in Kotest. Formerly, executors were not always shut down, which lead to leaking threads.)

My tendency is that each Kotest release should consider defaulting to the most common practice of its time. This might imply shifting towards dispatcher views by default in upcoming releases.

So I guess for now the above is one of the best options to run test scenarios as close to production as possible. An alternative would be to start using `Dispatchers.IO.limitedParallelism()` right away, but that's also marked as experimental. Other ideas on that?

Reply:
Definitely want to avoid experimental - I tried that in the past and it caused issues. I'd rather be conservative. This seems ok.